fix(transpiler): put pp. dirs on their own line no matter what the terminators are

AlexTMjugador opened the pull request:
Currently, when the formatting settings instruct the transpiler to use terminators that do not create newlines, it is possible for preprocessor directives to share a line with other external declarations, which violates the GLSL specification.

For example, given the following input:

```glsl
#version 110
vec3 u;
vec3 v;
#pragma "hi"
vec3 w;
```

The current transpiler would generate the following invalid output when `declaration_terminator` is set to `Whitespace::None`:

```glsl
#version 110
vec3 u;vec3 v;#pragma "hi"
vec3 w;
```

To fix this issue, let's make the formatting state capable of ensuring that the output cursor is at an empty line for preprocessor directives, and make preprocessor directives force a line flush to occur for the next external declaration.

While at it, I've fixed the else-if preprocessor directive incorrectly being transpiled to `#elseif` instead of `#elif`.

These changes depend on https://github.com/vtavernier/glsl-lang/pull/15 for the tests to pass with the v2-full lexer, as they expect such directives to be injected at the expected positions. Please let me know if you'd like for me to rebase these changes on top of those at some point :smile:

vtavernier commented:
Good catch! The fix looks good to me, the only thing left is to update the snapshots to reflect the changes (`env UPDATE_EXPECT=1 cargo test`). I do not have write access to the PR branch, so here's the change that's needed:

```diff
commit 0480cc2a6a10551f3b7a8c1ef622963940d9e655
Author: Vincent Tavernier <v.tavernier@pm.me>
Date:   Sun Apr 2 23:30:48 2023 +0200

    test(transpiler): update test snapshots

diff --git a/lang/src/transpiler/glsl.rs b/lang/src/transpiler/glsl.rs
index 8a8a48c5..87a39dde 100644
--- a/lang/src/transpiler/glsl.rs
+++ b/lang/src/transpiler/glsl.rs
@@ -2296,10 +2296,10 @@ return u;
         .unwrap();
 
         let expected = expect![[r#"
+            vec3 u;
             #version 110
-            vec3 u;vec3 v;
-            #pragma "hi"
-            vec3 w;"#]];
+            vec3 v;vec3 w;
+            #pragma "hi""#]];
 
         expected.assert_eq(&s);
     }
```

vtavernier reviewed (changes requested):
See https://github.com/vtavernier/glsl-lang/pull/21#issuecomment-1493445029

AlexTMjugador commented:
Thanks! I've updated the expected outputs in tests, but they didn't pass because they assumed that #15 was merged. That PR has been merged to the `next` branch now, so I rebased this PR on top of `next`, meaning that everything should be fine now :tada:
